prow/spyglass/lenses/metadata/lens.go

```diff
@@ -137,6 +146,79 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
 	return buf.String()
 }
 
+var failedMountRegex = regexp.MustCompile(`MountVolume.SetUp failed for volume "(.+?)" : (.+?)`)
+
+func hintFromPodInfo(buf []byte) string {
```

Comment:
Please unit test this function for the types of errors that are exposed and for happy cases like running and successful pods.

Reply:
Done.